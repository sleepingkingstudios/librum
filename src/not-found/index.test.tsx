import * as React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';

import { NotFoundPage } from './index';

describe('<NotFoundPage>', () => {
  it('should navigate back to the home page', () => {
    const { getByRole } = render(
      <MemoryRouter>
        <NotFoundPage />
      </MemoryRouter>
    );
    const link = getByRole('link', { name: 'Turn Back' });

    expect(link).toBeVisible();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<NotFoundPage />, { wrapper: MemoryRouter });

    expect(asFragment()).toMatchSnapshot();
  });
});
