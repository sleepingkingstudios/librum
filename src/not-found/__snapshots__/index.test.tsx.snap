// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-background dark:bg-background-dark flex flex-col max-w-screen-lg min-h-screen mx-auto p-3 text-text-primary"
  >
    <header
      class="font-serif leading-9 mb-3 text-5xl text-center sm:text-left text-text-strong dark:text-text-strong-dark"
    >
      Librum
      <br
        class="block sm:hidden"
      />
      <small
        class="m-0 sm:ml-2 text-2xl"
      >
        Campaign Companion
      </small>
      <hr
        class="fancy_hr border-text-strong after:border-text-strong dark:border-text-strong-dark dark:after:border-text-strong-dark mt-1"
      />
    </header>
    <main
      class="grow"
    >
      <article>
        <h1
          class="font-serif text-text-strong dark:text-text-strong-dark mb-2 text-3xl text-center sm:text-left"
        >
          Not Found
        </h1>
        <p
          class="mb-2 text-red-600 dark:text-red-900 text-center sm:text-left"
        >
          No more tales worth telling, no more lives worth living. The fire fades.
        </p>
        <p
          class="mb-2 text-red-600 dark:text-red-900 text-center sm:text-left"
        >
          Now there is only enough light to look upon yesterday and despair-
        </p>
        <p
          class="mb-2 text-red-600 dark:text-red-900 text-center sm:text-left"
        >
          For all that I make is unworthy of completion.
        </p>
        <hr
          class="fancy_hr border-muted after:border-muted dark:border-muted dark:after:border-muted m-5"
        />
        <p
          class="text-center sm:text-left text-xl"
        >
          <a
            class="font-serif text-red-600 hover:text-red-700 dark:text-red-900 dark:hover:text-red-800"
            href="/"
          >
            Turn Back
          </a>
        </p>
      </article>
    </main>
    <footer
      class="mt-3 shrink-0 text-center sm:text-left text-muted"
    >
      <hr
        class="fancy_hr border-text-strong after:border-text-strong dark:border-text-strong-dark dark:after:border-text-strong-dark mb-2"
      />
      What lies beyond the furthest reaches of the sky?
    </footer>
  </div>
</DocumentFragment>
`;
